src/app/views/datasets-new/edit-metadata-cantabular/CantabularMetadataController.jsx

```diff
@@ -0,0 +1,755 @@
+            fieldName === "contactTelephone" ||
+            fieldName === "relatedDatasets" ||
+            fieldName === "relatedPublication" ||
+            fieldName === "relatedMethodologies" ||
+            fieldName === "releaseFrequency" ||
+            fieldName === "unitOfMeasure" ||
+            fieldName === "qmi" ||
+            fieldName === "nextReleaseDate"
+        ) {
+            return true;
+        }
+        return this.state.datasetMetadataHasChanges;
+    };
+
+    versionMetadataHasChanges = fieldName => {
+        if (fieldName === "releaseDate" || fieldName === "notices" || fieldName === "usageNotes" || fieldName === "latestChanges") {
+            return true;
+        }
+        return this.state.versionMetadataHasChanges;
+    };
+
+    handleBackButton = () => {
+        const previousUrl = url.resolve("../../../");
+        this.props.dispatch(push(previousUrl));
+    };
+
+    mapMetadataToPutBody = (isSubmittingForReview, isMarkingAsReviewed) => {
+        return {
+            dataset: {
+                id: this.props.params.datasetID,
+                title: this.state.metadata.title,
+                description: this.state.metadata.summary,
+                keywords: this.state.metadata.keywords ? this.state.metadata.keywords.split(", ") : [],
+                national_statistic: this.state.metadata.nationalStatistic,
+                license: this.state.metadata.licence,
+                related_datasets: this.state.metadata.relatedDatasets,
+                publications: this.state.metadata.relatedPublications,
+                methodologies: this.state.metadata.relatedMethodologies,
+                qmi: {
+                    href: this.state.metadata.qmi,
+                },
+                release_frequency: this.state.metadata.releaseFrequency,
+                contacts: [
+                    {
+                        name: this.state.metadata.contactName,
+                        email: this.state.metadata.contactEmail,
+                        telephone: this.state.metadata.contactTelephone,
+                    },
+                ],
+                next_release: this.state.metadata.nextReleaseDate,
+                unit_of_measure: this.state.metadata.unitOfMeasure,
+            },
+            version: {
+                id: this.state.metadata.versionID,
+                release_date: this.state.metadata.releaseDate.value,
+                alerts: this.state.metadata.notices,
+                usage_notes: this.state.metadata.usageNotes,
+                lastest_changes: this.state.metadata.latestChanges,
+            },
+            dimensions: [...this.state.metadata.dimensions],
+            collection_id: this.props.params.collectionID,
+            collection_state: this.mapCollectionState(isSubmittingForReview, isMarkingAsReviewed),
+        };
+    };
+
+    mapCollectionState = (isSubmittingForReview, isMarkingAsReviewed) => {
+        const StatusInProgress = "InProgress";
```

Comment:
This may be inherited from the original in which case this remark is out of scope for this PR so please ignore it if that's the case, but ideally shouldn't states like this be defined in a single place for use throughout the application rather than defined locally and sent out into the application for consumption?